File: full-stack-asset-transfer-guide/applications/trader-typescript/src/commands/transfer.ts
/*
 * Copyright IBM Corp. All Rights Reserved.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { Gateway } from '@hyperledger/fabric-gateway';
import { CHAINCODE_NAME, CHANNEL_NAME } from '../config';
import { AssetTransfer } from '../contract';
import { assertDefined } from '../utils';

const usage = 'Arguments: <assetId> <ownerName> <ownerMspId>';

export default async function main(gateway: Gateway, args: string[]): Promise<void> {
    const assetId = assertDefined(args[0], usage);
    const newOwner = assertDefined(args[1], usage);
    const newOwnerOrg = assertDefined(args[2], usage);

    const network = gateway.getNetwork(CHANNEL_NAME);
    const contract = network.getContract(CHAINCODE_NAME);

    const smartContract = new AssetTransfer(contract);
    await smartContract.transferAsset(assetId, newOwner, newOwnerOrg);
}

File: full-stack-asset-transfer-guide/applications/trader-go/commands/transfer.go
package commands

import (
	"errors"

	"github.com/hyperledger/fabric-gateway/pkg/client"
)

// cmdTransfer transfers ownership of an asset to a new owner in a different organisation.
// Arguments: <assetId> <ownerName> <ownerMspId>
func cmdTransfer(gw *client.Gateway, args []string) error {
	if len(args) < 3 {
		return errors.New("arguments: <assetId> <ownerName> <ownerMspId>")
	}

	network := gw.GetNetwork(channelName())
	contract := network.GetContract(chaincodeName())

	smartContract := NewAssetTransfer(contract)
	return smartContract.TransferAsset(args[0], args[1], args[2])
}
